v2/internal/controllers/crd_redhatopenshift_openshiftcluster_20230401_test.go

```diff
@@ -0,0 +1,172 @@
+
+	roleAssingmentFromRPtoVnet := newRoleAssignment(tc, vnet, "rollassginmentrp", contributorRoleId)
+	roleAssingmentFromRPtoVnet.Spec.PrincipalId = to.Ptr(azureRedHadOpenshiftRPIdentityPrincipalId)
+
+	cluster := &aro.OpenShiftCluster{
+		ObjectMeta: tc.MakeObjectMeta("aro-cluster"),
+		Spec: aro.OpenShiftCluster_Spec{
+			Location: tc.AzureRegion,
+			Owner:    testcommon.AsOwner(rg),
+			ApiserverProfile: &aro.APIServerProfile{
+				Visibility: to.Ptr(aro.Visibility_Private),
+			},
+			ClusterProfile: &aro.ClusterProfile{
+				Domain:  to.Ptr("aro-example.com"),
+				Version: to.Ptr("4.14.16"),
+				// This is RG below is created by the RP.
+				ResourceGroupId:      to.Ptr(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", tc.AzureSubscription, tc.Namer.GenerateName("cluster-rg"))),
+				FipsValidatedModules: to.Ptr(aro.FipsValidatedModules_Disabled),
+			},
+			IngressProfiles: []aro.IngressProfile{
+				{
+					Name:       to.Ptr("default"),
+					Visibility: to.Ptr(aro.Visibility_Private),
+				},
+			},
+			MasterProfile: &aro.MasterProfile{
+				SubnetReference:  tc.MakeReferenceFromResource(masterSubnet),
+				VmSize:           to.Ptr("Standard_D8s_v3"),
+				EncryptionAtHost: to.Ptr(aro.EncryptionAtHost_Disabled),
+			},
+			NetworkProfile: &aro.NetworkProfile{
+				PodCidr:     to.Ptr("10.128.0.0/14"),
+				ServiceCidr: to.Ptr("172.30.0.0/16"),
+			},
+			ServicePrincipalProfile: &aro.ServicePrincipalProfile{
+				ClientId:     to.Ptr(details.clientId),
+				ClientSecret: &clientSecretRef,
+			},
+			WorkerProfiles: []aro.WorkerProfile{
+				{
+					Count:            to.Ptr(3),
+					Name:             to.Ptr("worker"),
+					SubnetReference:  tc.MakeReferenceFromResource(workerSubnet),
+					VmSize:           to.Ptr("Standard_D4s_v3"),
+					DiskSizeGB:       to.Ptr(128),
+					EncryptionAtHost: to.Ptr(aro.EncryptionAtHost_Disabled),
+				},
+			},
+		},
+	}
+
+	tc.CreateResourcesAndWait(
+		vnet,
+		roleAssingmentToVnet,
+		roleAssingmentFromRPtoVnet,
+		workerSubnet,
+		masterSubnet,
+		cluster)
+
+	tc.Expect(cluster.Status.Id).ToNot(BeNil())
+	armId := *cluster.Status.Id
+	tc.Expect(armId).ToNot(BeNil())
+
+	tc.DeleteResourceAndWait(cluster)
+
+	// Ensure that the resource was really deleted in Azure
+	exists, retryAfter, err := tc.AzureClient.CheckExistenceWithGetByID(tc.Ctx, armId, string(aro.APIVersion_Value))
+	tc.Expect(err).ToNot(HaveOccurred())
+	tc.Expect(retryAfter).To(BeZero())
+	tc.Expect(exists).To(BeFalse())
+
+}
+
+func getIds(tc *testcommon.KubePerTestContext) servicePrincipalDetails {
+	nilUUID := uuid.Nil.String()
+
+	clientId := nilUUID
+	objectId := nilUUID
+	clientSecret := tc.Namer.GenerateSecretOfLength(40)
+
+	if id := os.Getenv("ARO_CLIENT_ID"); id != "" {
+		clientId = id
+	}
+
+	if id := os.Getenv("ARO_OBJECT_ID"); id != "" {
+		objectId = id
+	}
+
+	if secret := os.Getenv("ARO_CLIENT_SECRET"); secret != "" {
+		clientSecret = secret
+	}
+
+	tc.WithLiteralRedaction(clientId, nilUUID)
```

Comment:
I think you don't want to add these redactions if there's nothing to redact (input is 0000's already), as it's causing the tests to fail.